File: py/utility/make_error.py
# FuelPricesAtSpanishGasStations SDK utility: make_error

from __future__ import annotations
from core.operation import FuelPricesAtSpanishGasStationsOperation
from core.result import FuelPricesAtSpanishGasStationsResult
from core.control import FuelPricesAtSpanishGasStationsControl
from core.error import FuelPricesAtSpanishGasStationsError


def make_error_util(ctx, err):
    if ctx is None:
        from core.context import FuelPricesAtSpanishGasStationsContext
        ctx = FuelPricesAtSpanishGasStationsContext({}, None)

    op = ctx.op
    if op is None:
        op = FuelPricesAtSpanishGasStationsOperation({})
    opname = op.name
    if opname == "" or opname == "_":
        opname = "unknown operation"

    result = ctx.result
    if result is None:
        result = FuelPricesAtSpanishGasStationsResult({})
    result.ok = False

    if err is None:
        err = result.err
    if err is None:
        err = ctx.make_error("unknown", "unknown error")

    errmsg = ""
    if isinstance(err, FuelPricesAtSpanishGasStationsError):
        errmsg = err.msg
    elif hasattr(err, "msg") and err.msg is not None:
        errmsg = err.msg
    elif isinstance(err, str):
        errmsg = err
    else:
        errmsg = str(err)

    msg = "FuelPricesAtSpanishGasStationsSDK: " + opname + ": " + errmsg
    msg = ctx.utility.clean(ctx, msg)

    result.err = None

    spec = ctx.spec

    if ctx.ctrl.explain is not None:
        ctx.ctrl.explain["err"] = {"message": msg}

    sdk_err = FuelPricesAtSpanishGasStationsError("", msg, ctx)
    sdk_err.result = ctx.utility.clean(ctx, result)
    sdk_err.spec = ctx.utility.clean(ctx, spec)

    if isinstance(err, FuelPricesAtSpanishGasStationsError):
        sdk_err.code = err.code

    ctx.ctrl.err = sdk_err

    if ctx.ctrl.throw_err is False:
        return result.resdata, None

    return None, sdk_err

File: .sdk/tm/php/utility/MakeError.php
<?php
declare(strict_types=1);

// FuelPricesAtSpanishGasStations SDK utility: make_error

require_once __DIR__ . '/../core/Operation.php';
require_once __DIR__ . '/../core/Result.php';
require_once __DIR__ . '/../core/Error.php';

class FuelPricesAtSpanishGasStationsMakeError
{
    public static function call(?FuelPricesAtSpanishGasStationsContext $ctx, mixed $err): array
    {
        if ($ctx === null) {
            require_once __DIR__ . '/../core/Context.php';
            $ctx = new FuelPricesAtSpanishGasStationsContext([], null);
        }
        $op = $ctx->op ?? new FuelPricesAtSpanishGasStationsOperation([]);
        $opname = $op->name;
        if ($opname === '' || $opname === '_') {
            $opname = 'unknown operation';
        }

        $result = $ctx->result ?? new FuelPricesAtSpanishGasStationsResult([]);
        $result->ok = false;

        if ($err === null) {
            $err = $result->err;
        }
        if ($err === null) {
            $err = $ctx->make_error('unknown', 'unknown error');
        }

        $errmsg = ($err instanceof FuelPricesAtSpanishGasStationsError) ? $err->msg : (string)$err;
        $msg = "FuelPricesAtSpanishGasStationsSDK: {$opname}: {$errmsg}";
        $msg = ($ctx->utility->clean)($ctx, $msg);

        $result->err = null;
        $spec = $ctx->spec;

        if ($ctx->ctrl->explain) {
            $ctx->ctrl->explain['err'] = ['message' => $msg];
        }

        $sdk_err = new FuelPricesAtSpanishGasStationsError('', $msg, $ctx);
        $sdk_err->result = ($ctx->utility->clean)($ctx, $result);
        $sdk_err->spec = ($ctx->utility->clean)($ctx, $spec);
        if ($err instanceof FuelPricesAtSpanishGasStationsError) {
            $sdk_err->sdk_code = $err->sdk_code;
        }

        $ctx->ctrl->err = $sdk_err;

        if ($ctx->ctrl->throw_err === false) {
            return [$result->resdata, null];
        }
        return [null, $sdk_err];
    }
}
